src/lib/util/cdrom.cpp

```diff
@@ -1625,6 +1718,33 @@ bool ecc_verify(const uint8_t *sector)
 	return true;
 }
 
+/**
+ * @fn  void edc_generate(uint8_t *sector)
+ *
+ * @brief   -------------------------------------------------
+ *            edc_generate - generate the EDC checksum for a sector, overwriting any
+ *            existing checksum, code is only valid for MODE1 sectors
+ *          -------------------------------------------------.
+ *
+ * @param [in,out]  sector  If non-null, the sector.
+ */
+void edc_generate(uint8_t *sector)
+{
+	uint32_t result = 0;
+	for (int byte = 0; byte < EDC_MODE1_OFFSET; byte += 4)
+	{
+		result = EDC_crctable[(result ^ sector[byte+0]) & 0xffL] ^ (result >> 8);
+		result = EDC_crctable[(result ^ sector[byte+1]) & 0xffL] ^ (result >> 8);
+		result = EDC_crctable[(result ^ sector[byte+2]) & 0xffL] ^ (result >> 8);
+		result = EDC_crctable[(result ^ sector[byte+3]) & 0xffL] ^ (result >> 8);
+	}
+
+	sector[EDC_MODE1_OFFSET+0] = ((uint8_t *) &result)[0];
+	sector[EDC_MODE1_OFFSET+1] = ((uint8_t *) &result)[1];
+	sector[EDC_MODE1_OFFSET+2] = ((uint8_t *) &result)[2];
+	sector[EDC_MODE1_OFFSET+3] = ((uint8_t *) &result)[3];
```

Comment:
Type punning like this is generally a bad idea and causes endian issues, particulrly when you can just use shits.

Reply:
resolved in b08d62705f5c29a1753392b7ca7353d777c02f7a